satellite/console/service.go

```diff
@@ -125,10 +125,16 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	}
 	if user.PartnerID != "" {
 		offerType = rewards.Partner
+	} else if refUserID != "" {
+		offerType = rewards.Referral
 	}
 
 	//TODO: Create a current offer cache to replace database call
-	currentReward, err := s.rewards.GetCurrentByType(ctx, offerType)
+	offers, err := s.rewards.GetActiveOffersByType(ctx, offerType)
+	if err != nil && !rewards.NoCurrentOfferErr.Has(err) {
+		return nil, errs.New(internalErrMsg)
```

Comment:
this is an internal error we have to log it for having a clue what could be wrong.

Add this line before returning `s.log.Error("internal error", zap.Error(err))`

I would say that's worth to do in this line 139 too, but that line has not been modified in this branch so it could be done another time.